src/physics/base/ParticleMd.hh

```diff
@@ -0,0 +1,60 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file ParticleMd.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <string>
+#include "base/OpaqueId.hh"
+
+namespace celeritas
+{
+struct ParticleMd;
+using PDGNumber = OpaqueId<ParticleMd, int>;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Host-only metadata for a particle type.
+ *
+ * The PDG Monte Carlo number is a unique "standard model" identifier for a
+ * particle. See "Monte Carlo Particle Numbering Scheme" in the "Review of
+ * Particle Physics":
+ * http://pdg.lbl.gov/2019/reviews/rpp2019-rev-monte-carlo-numbering.pdf
+ *
+ */
+struct ParticleMd
+{
+    std::string name;     // Particle name
+    PDGNumber   pdg_code; // See "Review of Particle Physics"
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * \namespace pdg
+ *
+ * Unique standard model particle identifiers by the Particle Data Group.
+ *
+ * These should be extended as needed.
+ */
+namespace pdg
+{
+//---------------------------------------------------------------------------//
+//@{
+//! Particle Data Group Monte Carlo number codes.
+#define CELER_DEFINE_PDGNUMBER(NAME, VALUE) \
+    inline constexpr PDGNumber NAME() { return PDGNumber{VALUE}; }
+
+CELER_DEFINE_PDGNUMBER(electron, 11)
+CELER_DEFINE_PDGNUMBER(gamma, 22)
+CELER_DEFINE_PDGNUMBER(neutron, 2112)
+
+#undef CELER_DEFINE_PDGNUMBER
```

Comment:
Now that we are going to have all particle info imported, do we still want to keep a macro for pdg and name?

Reply:
Yes, I think that all the fundamental particles for which we have implemented processes should have their PDG numbers hard-coded here. This way, we'll get a compile error if we mistype an identifier rather than a runtime error, and the code will be more readable to people who don't have the PDG numbers hard-coded. 

For example, the KN interactor data requires particle def IDs for electron and gamma. With this technique, we can write:
```c++
        pointers_.electron_id = params.find(pdg::electron());
```
instead of
```c++
        pointers_.electron_id = params.find(11);
```
which is not readable (unless you've got the numeric codes memorized), or
```c++
        pointers_.electron_id = params.find("electron");
```
which is obviously fragile since it seems geant defines it as `"e-"` rather than `"Electron"` or `"electron"` or `"E-"` etc. 😉

Reply:
Ok, these two (compile error + solid function call) are pretty compelling reasons.